### PR TITLE
MAINT: Use legacy pip/setuptools for py2 checking

### LINKED ISSUE #2154: [CI] Build failure
```
#!/bin/bash -eo pipefail
mkdir -p /tmp/${DATASET}/derivatives
sudo setfacl -d -m group:$(id -gn):rwx /tmp/${DATASET}/derivatives
sudo setfacl -m group:$(id -gn):rwx /tmp/${DATASET}/derivatives
pip install --upgrade pip setuptools
pip install --upgrade /tmp/src/fmriprep/wrapper/

Collecting pip
  Downloading https://files.pythonhosted.org/packages/43/84/23ed6a1796480a6f1a2d38f2802901d078266bda38388954d01d3f2e821d/pip-20.1.1-py2.py3-none-any.whl (1.5MB)
    100% |████████████████████████████████| 1.5MB 779kB/s 
Collecting setuptools
  Downloading https://files.pythonhosted.org/packages/25/bf/a37e89d3148221fedd4def62bb68749041d79f3840d58a7943f81a6f6c6e/setuptools-47.1.1.zip (870kB)
    100% |████████████████████████████████| 870kB 1.3MB/s 
    Complete output from command python setup.py egg_info:
    pkg_resources/py2_warn.py:15: UserWarning: Setuptools no longer works on Python 2
    ************************************************************
    Encountered a version of Setuptools that no longer supports
    this version of Python. Please head to
    https://bit.ly/setuptools-py2-sunset for support.
    ************************************************************
      warnings.warn(pre + "*" * 60 + msg + "*" * 60)
```

we probably need to either:
1) remove python 2 from wrapper testing
2) downgrade setuptools

### PR DESCRIPTION
**Pull request by mgxd**
Closes #2154

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -276,7 +276,7 @@ jobs:
             pyenv local $PY2
             echo -n "Python version: "
             python --version
-            pip install --upgrade pip setuptools
+            pip install --upgrade "pip<21" "setuptools<45"
```
Comment (effigies) [inline .circleci/config.yml:279]: :+1:

File: .circleci/config.yml [hunk 2]
```diff
@@ -395,7 +395,7 @@ jobs:
             mkdir -p /tmp/${DATASET}/derivatives
             sudo setfacl -d -m group:$(id -gn):rwx /tmp/${DATASET}/derivatives
             sudo setfacl -m group:$(id -gn):rwx /tmp/${DATASET}/derivatives
-            pip install --upgrade pip setuptools
+            pip install --upgrade "pip<21" "setuptools<45"
             pip install --upgrade /tmp/src/fmriprep/wrapper/
```
Comment (effigies) [inline .circleci/config.yml:399]: Can we assume that these images have Python 3 as well? If so, I think probably the cleaner long-term solution is this:

```suggestion
            python3 -m pip install --upgrade pip setuptools
            python3 -m pip install --upgrade /tmp/src/fmriprep/wrapper/
```

Keeping the Py2 accommodations to be both explicit and minimal is preferable.

**Comment by oesteban** [comment]
I think we can replace the old classic images with the new convenience images:
https://app.circleci.com/pipelines/github/oesteban/fmriprep/126/workflows/af18815e-4149-41e0-9f38-42589bd9602c

**Comment by oesteban** [comment]
@mgxd please check https://github.com/mgxd/fmriprep/pull/2 (a0907c2). You'll need to roll back your latest commit (just to keep history clean).

**Pull request merged**